From 1a9a6a5adaae030f5a454435362674ebb5affa6c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Tue, 4 Feb 2014 14:28:22 +0100
Subject: [PATCH] force fbdev to use gl by default
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Andreas Müller <schnitzeltony@googlemail.com>
---
 src/compositor-fbdev.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/compositor-fbdev.c b/src/compositor-fbdev.c
index 0d96269..53bc0dd 100644
--- a/src/compositor-fbdev.c
+++ b/src/compositor-fbdev.c
@@ -963,7 +963,7 @@ backend_init(struct wl_display *display, int *argc, char *argv[],
 	struct fbdev_parameters param = {
 		.tty = 0, /* default to current tty */
 		.device = "/dev/fb0", /* default frame buffer */
-		.use_gl = 0,
+		.use_gl = 1,
 	};
 
 	const struct weston_option fbdev_options[] = {
-- 
1.8.3.1

