From 01ff96f8616c3d07d30d1943cdd5259e99b92f3a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Fri, 16 May 2014 11:20:21 +0200
Subject: [PATCH] omap3630/sysconfig.c: add debug information in SysInitialise
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Müller <schnitzeltony@googlemail.com>
---
 services4/system/omap3630/sysconfig.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/services4/system/omap3630/sysconfig.c b/services4/system/omap3630/sysconfig.c
index f38292e..4483bcf 100644
--- a/services4/system/omap3630/sysconfig.c
+++ b/services4/system/omap3630/sysconfig.c
@@ -322,6 +322,7 @@ PVRSRV_ERROR SysInitialise(IMG_VOID)
 #if !defined(SGX_DYNAMIC_TIMING_INFO)
 	SGX_TIMING_INFORMATION*	psTimingInfo;
 #endif
+	PVR_TRACE(("SysInitialise started"));
 	gpsSysData = &gsSysData;
 	OSMemSet(gpsSysData, 0, sizeof(SYS_DATA));
 
@@ -529,6 +530,7 @@ PVRSRV_ERROR SysInitialise(IMG_VOID)
 	}
 #endif /* !defined(PVR_NO_OMAP_TIMER) */
 
+	PVR_TRACE(("SysInitialise OK"));
 
 	return PVRSRV_OK;
 }
-- 
1.8.3.1

