From bfdd63ee539a77715d0cbd57a0839b6eb85f1329 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Fri, 17 Nov 2023 18:53:49 +0100
Subject: [PATCH] Reduce bootdelay to 1s
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>
---
 include/configs/omap3_overo.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/omap3_overo.h b/include/configs/omap3_overo.h
index e49e6b23c3..d841df1121 100644
--- a/include/configs/omap3_overo.h
+++ b/include/configs/omap3_overo.h
@@ -144,7 +144,7 @@
 							/* partition */
 
 /* Environment information */
-#define CONFIG_BOOTDELAY		5
+#define CONFIG_BOOTDELAY		1
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"loadaddr=0x82000000\0" \
-- 
2.41.0

